autoload/deoplete/handler.vim

```diff
@@ -73,13 +73,14 @@ function! s:do_complete(timer) abort
   endif
 endfunction
 
-function! deoplete#handler#_completion_timer_start() abort
+function! s:completion_timer_start(event) abort
   if exists('s:completion_timer')
     call s:completion_timer_stop()
   endif
 
   let delay = max([20, g:deoplete#auto_complete_delay])
-  let s:completion_timer = timer_start(delay, function('s:do_complete'))
+  " TODO: wrapper instead of lambda?!
+  let s:completion_timer = timer_start(delay, {-> s:completion_begin(a:event)})
```

Comment:
@Shougo 
I assume we cannot use lambdas here?

I will happily adjust this then, awaiting your general feedback first.